rewrite-xml/src/main/java/org/openrewrite/xml/tree/Xml.java

```diff
@@ -162,6 +167,28 @@ public <S, T extends S> T service(Class<S> service) {
             }
             return SourceFile.super.service(service);
         }
+
+        @Nullable
+        @NonFinal
+        SoftReference<JavaTypeReferences> javaTypeReferences;
```

Comment:
Not sure how important that `@Transient` on the getter is, but we should definitely have a `transient` modifier here.

```suggestion
        transient SoftReference<JavaTypeReferences> javaTypeReferences;
```